Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -157,21 +190,22 @@ CGPathRef CGPathCreateCopy(CGPathRef path) {
 @Status Interoperable
 */
 CGMutablePathRef CGPathCreateMutableCopy(CGPathRef path) {
-    if (path == NULL) {
-        return NULL;
-    }
+    RETURN_NULL_IF(!path);
 
     CGMutablePathRef mutableRet = CGPathCreateMutable();
 
     // In order to call stream and copy the contents of the original path into the
     // new copy we must close this path.
     // Otherwise the D2D calls will return that a bad state has been entered.
-    path->closePath();
+    path->ClosePath();
 
-    FAIL_FAST_IF_FAILED(path->_impl.pathGeometry->Stream(mutableRet->_impl.geometrySink.Get()));
+    if (FAILED(path->PathGeometry()->Stream(mutableRet->GeometrySink().Get()))) {
+        CGPathRelease(mutableRet);
```

Comment:
>CGPathRelease(mutableRet); [](start = 8, length = 26)

seems like an RAII construct may be useful for ensuring resources are properly released?